src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -3275,24 +3289,35 @@ void GraphOptimizer::DropRedundantMemoryOutput(Graph &graph) {
         graph.RemoveEdge(memoryOutputNode->getParentEdgeAt(0));
         // there are no output edges from MemoryOutput nodes
 
+        CPU_GRAPH_OPTIMIZER_SCOPE(DropRedundantMemoryOutput_SubGraph);
+        auto memInpNd = std::dynamic_pointer_cast<node::MemoryInput>(node);
```

Comment:
I am a bit confused here.
`auto subGraph = memInpNd ? memInpNd->getSubGraph() : nullptr;`
If `memInpNd`  is empty, subGraph will be nullptr. And `MemoryInputSingle` can accept empty subgraph. memInpNd also only use here.

Reply:
Yes. But it makes sense to introduce a sanity check that the is indeed of type `node::MemoryInput`. Because if `memInpNd` is null doesn't mean there is no subgraph, but rather that the whole further execution doesn't make sense and this is an exceptional situation.

Reply:
Added:
`OPENVINO_ASSERT(memInpNd, "MemoryInput node ", node->getName(), " has unexpected dynamic type");`